### PR TITLE
fix: EDFM bugfixes, part of #3348

### PR DESCRIPTION
**Pull request by paveltomin**
part of https://github.com/GEOS-DEV/GEOS/pull/3348/
1. derivatives sign fix
2. frac/cell element volume
3. replace FaceElementSubRegion by SurfaceElementSubRegion in SinglePhaseBase
4. fix inconsistency between default aperture and hydraulic aperture table in EDFM test cases

### CODE REVIEW

**Review by Guotong-Ren — APPROVED** [review]

File: src/coreComponents/constitutive/contact/CoulombFriction.hpp [hunk 1]
```diff
@@ -265,6 +265,8 @@ inline void CoulombFrictionUpdates::computeShearTraction( localIndex const k,
       real64 dLimitTau_dNormalTraction;
       real64 const limitTau = computeLimitTangentialTractionNorm( tractionVector[0],
                                                                   dLimitTau_dNormalTraction );
+      // dLimitTau_dNormalTraction from the function above has a wrong sign, flip it
+      dLimitTau_dNormalTraction = -dLimitTau_dNormalTraction;
```
Comment (CusiniM) [inline src/coreComponents/constitutive/contact/CoulombFriction.hpp]: why not fixing it inside the function?

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/constitutive/contact/CoulombFriction.hpp (hunk 1 above)
Comment (paveltomin) [reply src/coreComponents/constitutive/contact/CoulombFriction.hpp]: the functions is being used also by `SolidMechanicsLagrangeContact`, just limiting changes to EDFM here

Comment (CusiniM) [reply src/coreComponents/constitutive/contact/CoulombFriction.hpp]: but the sign should not change. I mean I would expect that derivative to be the same

**Review by CusiniM — APPROVED** [review]

**Comment by paveltomin** [comment]
checked the tests diffs and everything looks good - only edfm cases are different as expected, improvements in nonlinear convergence as expected

**Pull request merged**